File: leetcode/349-intersection-of-two-arrays.py
# https://leetcode.com/problems/intersection-of-two-arrays/

from typing import List

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        num1_set = set(nums1)
        num2_set = set(nums2)

        intersection = []
        for num in num1_set:
            if num in num2_set:
                intersection.append(num)
        
        return intersection

File: leetcode/349-intersection-of-two-arrays.js
// https://leetcode.com/problems/intersection-of-two-arrays/

/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
  const nums1Set = new Set(nums1);
  const nums2Set = new Set(nums2);

  const intersection = [];

  nums1Set.forEach((num) => {
    if (nums2Set.has(num)) intersection.push(num);
  });

  return intersection;
};
